### PR TITLE
philipmiesbauer's Vi Cheatsheet Goodie

### PR DESCRIPTION
**Pull request by philipmiesbauer**
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
Cheatsheet

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Cheatsheet for Vi

**What is the data source for your Instant Answer? (Provide a link if possible)**
Lagmonster, http://www.lagmonster.org/docs/vi.html

**Why did you choose this data source?**
Cause I looked for it several times now and was always sad, that it wasn't there yet

**Are there any other alternative (better) data sources?**
maybe, but I usually used that one

**What are some example queries that trigger this Instant Answer?**
cheatsheet vi, vi cheatsheet,...

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
nerds, programmers, embedded systems programmers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
I don't know

**Which existing Instant Answers will this one supercede/overlap with?**
cheatsheet vim, but vi is farther known and simpler and available on more environments

**Are you having any problems? Do you need our help with anything?**
No, thanks

**Where did you hear about DuckDuckHack? (For first time contributors)**
a friend showed me duckduckgo and I searched till I found a place to contribute

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screenshot from 2015-08-14 15 02 41

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Linux- Ubuntu
        - [X]Firefox

    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

-----
IA Page: https://duck.co/ia/view/vi_cheat_sheet

### CODE REVIEW

**Comment by philipmiesbauer** [comment]
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
Cheatsheet

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Cheatsheet for Vi

**What is the data source for your Instant Answer? (Provide a link if possible)**
Lagmonster, http://www.lagmonster.org/docs/vi.html

**Why did you choose this data source?**
Cause I looked for it several times now and was always sad, that it wasn't there yet

**Are there any other alternative (better) data sources?**
maybe, but I usually used that one

**What are some example queries that trigger this Instant Answer?**
cheatsheet vi, vi cheatsheet,...

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
nerds, programmers, embedded systems programmers

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
I don't know

**Which existing Instant Answers will this one supercede/overlap with?**
cheatsheet vim, but vi is farther known and simpler and available on more environments

**Are you having any problems? Do you need our help with anything?**
No, thanks

**Where did you hear about DuckDuckHack? (For first time contributors)**
a friend showed me duckduckgo and I searched till I found a place to contribute

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
cheatsheet

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Linux- Ubuntu
        - [X]Firefox

    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

```

**Comment by philipmiesbauer** [comment]
screenshot from 2015-08-14 15 02 41

**Comment by moollaza** [comment]
@philipmiesbauer thanks a lot for contributing, we really appreciate it! We'll be sure to review this shortly and give you some feedback!

**Comment by moollaza** [comment]
@philipmiesbauer it seems like the source contains some section that don't exist in your Cheat Sheet. Any reason for that?

**Comment by philipmiesbauer** [comment]
I felt like the source is to extensive for a instant answer cheat sheet and
that people who would most likely look for a cheatsheet for vi are probably
not going to use the other functions and it would be rather confusing, but
if the want more, then the source link is still available.

**Comment by philipmiesbauer** [comment]
Should I add the rest of the data to the cheatsheet?

**Comment by moollaza** [comment]
@philipmiesbauer If you think any of the remaining data is useful and could fall under one of the existing groups, or a new one please add it. If this is good enough then I'm happy :+1: 

I'm sure the community will add more if they feel it's necessary.

File: share/goodie/cheat_sheets/vi.json [hunk 1]
```diff
@@ -0,0 +1,285 @@
+{
+    "id": "vi_cheat_sheet", 
+    "name": "Vi",
+    "description": "Awesome Text Editor",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/vi.json]: Although this is a great description, I think something a little more descriptive would be appropriate ;)

Comment (moollaza) [inline share/goodie/cheat_sheets/vi.json]: If there's nothing to say, we can remove the description altogether.

**Comment by moollaza** [comment]
@philipmiesbauer on second thought, this is good for now. We can add more later if needed. Thanks again for contributing, we really appreciate it!

**Pull request merged**